pkg/server/v3/delta_test.go

```diff
@@ -457,3 +458,95 @@ func TestDeltaCallbackError(t *testing.T) {
 		})
 	}
 }
+
+// BENCHMARKS =====================================================================================================
+
+func BenchmarkDeltaResponseHandlers(b *testing.B) {
+	for _, typ := range testTypes {
+		b.Run(typ, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				config := makeMockConfigWatcher()
+				config.deltaResponses = makeDeltaResponses()
+				s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+				resp := makeMockDeltaStream(&testing.T{})
+				// This is a wildcard request since we don't specify a list of resource subscriptions
+				res, err := config.deltaResponses[typ][0].GetDeltaDiscoveryResponse()
+				if err != nil {
+					b.Error(err)
+				}
+				// We only subscribe to one resource to see if we get the appropriate number of resources back
+				resp.recv <- &discovery.DeltaDiscoveryRequest{Node: node, TypeUrl: typ, ResourceNamesSubscribe: []string{res.Resources[0].Name}}
+
+				go func() {
+					err := process(typ, resp, s)
+					assert.NoError(&testing.T{}, err)
```

Comment:
maing a new testing.T isn't going to do the right thing i think

Reply:
Yea I actually just changed that to `b.Error` so I got rid of the assertion